doc/developer/guide-changes.md

```diff
@@ -0,0 +1,686 @@
+feedback early and often. If designing the details will take a week, spend a day
+writing up the high-level design and get feedback on _that_ first.
+
+As with many parts of the job, knowing when to seek feedback relies quite a bit
+on intuition. As you gain more experience with an area of the codebase, you'll
+be able to predict what kind of feedback you'll receive and preemptively address
+it, meaning you can embark on larger projects while seeking less up-front
+feedback. But you must still obey the [PR size limits](#pr-size-limits) for the
+sake of your reviewer!
+
+### Choosing reviewers
+
+At the moment, choosing reviewers is a bit of a dark art. The best reviewer for
+a PR is the person besides you who is most familiar with the code that is
+changed. If you are not sure who that is, ask in #engineering.
+
+Coming soon: a breakdown of which tech lead is responsible for each area of the
+codebase. The tech lead needn't review every PR in their area of responsibility,
+but can help direct code review requests to the appropriate engineer.
+
+### Merging
+
+#### When to merge
+
+For sufficiently trivial changes, you can consider merging without review. This
+should be somewhat rare, however, and only when you have high confidence that a)
+the change is correct, and b) the change is uncontroversial. When in doubt, get
+a review!
+
+When a reviewer approves your PR, they will either press the "approve" button
+in GitHub's code review interface, leaving a big green checkmark on your PR,
+or say something like "LGTM" ("looks good to me"). If they leave no other
+comments, then you are cleared to merge your PR!
+
+Sometimes a reviewer will leave a conditional approval. This is usually
+explicitly spelled out, as in "LGTM if you make the fix to the `foo` function;
+all the other comments are optional." If it is not clear whether an approval
+is conditional on addressing other comments in the review, err on the side of
+caution and consider them all blocking comments unless you confirm otherwise
+with your reviewer.
+
+On large PRs, you may receive only partial approvals, e.g., "the changes to the
+SQL layer LGTM, but please get someone else to review the docs changes." In this
+case you must use your discretion as to when you have received a covering set
+of approvals. (This is a great reason to prefer small PRs when possible, since
+you can gather full approvals from various owners in parallel.)
+
+#### Merge skew
+
+Sometimes, two PRs are fine when tested isolation, but the combination of the
+two PRs landing together on `main` can cause tests to fail—or even cause
+Materialize to fail to compile entirely. We call this situation "merge skew."
+
+When this happens, Buildkite will send a Slack message to #engineering
+indicating that tests failed on `main`. If your PR is implicated in the failure,
+please try to jump on fixing the merge skew as soon as possible, or nominate
+someone else to fix it if you are otherwise occupied.
```

Comment:
So, part of this comes my experience at Pure Storage, where `merge skew` was not the only cause for a broken `main`. There were a lot of long-running tests that could break and people would often "nominate someone else" without taking responsibility for fixing the test urgently. Hence why I believe it's important to escalate instead of nominate. Sending a message on Slack to say "Hey, I can't look now, but I'll look once available" works for me too.

If you believe that we can always run 100% of our tests on PRs before they merge, then I agree with what you've written. `merge skew` is sufficiently rare, and we are sufficiently small, that engineers will almost never by blocked by it.

Reply:
A note about bors use at Cockroach (currently in the many 10s of engineers): it was indeed suspected that bors caused more problems than it solved, so it was disabled. During the next day 5 merge skews occurred in master and it was thus re-enabled because it was evident that it was useful. I suggest having some data-backed trigger, like X merge skews per day/week and we look into using bors.

Reply:
Oh, that's helpful context, thanks Matt! Just spitballing, but merge skew once a week seems like about the right time to use Bors? I put that in for now, but definitely subject to debate.

Reply:
Yes? I think it makes sense to use bors if (time that bors saves preventing merge skews and its associated problems) > (time it takes to implement and perform maintenance on the bors service). 1/week might be there? Really hard to say.